spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1437,7 +1437,7 @@ object QueryPlanSerde extends Logging with CometExprShim {
 
       case s: StringDecode =>
         // Right child is the encoding expression.
-        s.right match {
+        s.charset match {
```

Comment:
nit: Is there a subtle difference in Spark 4? `s.bin` and `s.charset` are simply aliases for `s.left` and `s.right`
(I admit this looks more useful)